eng/common/pipelines/templates/steps/cache-ps-modules.yml

```diff
@@ -0,0 +1,10 @@
+steps:
+- pwsh: |
+    . ../../scripts/Helpers/PSModule-Helpers.ps1
+    Write-Host "##vso[task.setvariable variable=CachedPSModulePath]$global:CurrentUserModulePath"
+  displayName: Set PS Modules Cache Directory
+- task: Cache@2
+  inputs:
+    key: 'PSModulePath | "$(Agent.OS)"'
+    path: $(CachedPSModulePath)
```

Comment:
So the cache will end up being per OS + a directory like `/usr/share/<ps modules segment>/5.2/` ?

Reply:
Yes it will will be limited to OS because we don't really have any other good files to key off of currently to know when versions change.

For the path it will be the user path defined at https://docs.microsoft.com/en-
us/powershell/module/microsoft.powershell.core/about/about_psmodulepath which descripts